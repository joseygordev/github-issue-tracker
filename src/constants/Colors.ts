const tintColorLight = '#2f95dc';
const tintColorDark = '#fff';

export default {
  black: '#14181c',
  steel: '#414d5b',
  border: '#aab2ba',
  green: '#16c65d',
  blue: '#00aaff',
  red: '#ea1f59',
  yellow: '#ffe18b',
  pageBackground: '#f4f6f6',
  white: '#fff',
  light: {
    text: '#000',
    background: '#fff',
    tint: tintColorLight,
    tabIconDefault: '#ccc',
    tabIconSelected: tintColorLight
  },
  dark: {
    text: '#fff',
    background: '#000',
    tint: tintColorDark,
    tabIconDefault: '#ccc',
    tabIconSelected: tintColorDark
  },
};
