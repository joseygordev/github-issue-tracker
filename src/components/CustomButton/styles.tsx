import { StyleSheet } from 'react-native';
import Colors from '../../constants/Colors';

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.white,
    borderRadius: 15,
    width: 130,
    height: 42,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 8,
  },
  primaryButton: {
    width: 120,
    borderWidth: 1,
    borderColor: Colors.blue,
    backgroundColor: 'transparent',
  },
  secondaryButton: {
    width: 'auto',
    paddingHorizontal: 10,
  },
  defaultText: {
    color: Colors.blue,
    fontSize: 12,
    letterSpacing: 0.8,
  },
});
