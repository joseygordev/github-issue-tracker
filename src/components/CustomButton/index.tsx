import React from 'react';
import { TouchableOpacity, TouchableOpacityProps } from 'react-native';

import Colors from '../../constants/Colors';

import CustomText from '../CustomText';

import styles from './styles';

export default function CustomButton(props: TouchableOpacityProps & ButtonProps) {
  const { 
    style,
    type = 'default',
    text,
    disabled,
    leftIcon,
    rightIcon,
} = props;

  const buttonStyle = [
    styles.container,
    styles[(type + 'Button') as keyof ButtonStyle],
    disabled ? { borderColor: Colors.border } : {},
    style,
  ];

  const buttonTextStyle = [styles.defaultText, disabled ? { color: Colors.border } : {}];

  return (
    <TouchableOpacity {...props} style={buttonStyle}>
      {leftIcon}
      {Boolean(text) && <CustomText style={buttonTextStyle}>{text.toUpperCase()}</CustomText>}
      {rightIcon}
    </TouchableOpacity>
  );
}
export interface ButtonProps {
  text?: String;
  type?: 'primary' | 'secondary';
  leftIcon?: JSX.Element;
  rightIcon?: JSX.Element;
}
interface ButtonStyle {
  primaryButton: object;
  secondaryButton: object;
}
