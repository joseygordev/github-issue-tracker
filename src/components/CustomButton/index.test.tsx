import React from 'react';
import CustomButton from './index';
import { create } from 'react-test-renderer';
import Colors from '../../constants/Colors';
import { AntDesign } from '@expo/vector-icons';

describe('CustomButton', () => {
  it('it renders button correctly', async () => {
    const tree = create(<CustomButton testID="viewIssuesButton" type="primary" text="Button" onPress={() => {}} />);
  
    const textComponent = tree.root.findByProps({ testID: 'viewIssuesButton' }).props;
  
    expect(textComponent.type).toEqual('primary');
  });
  
  it('it renders button disabled', async () => {
    const tree = create(<CustomButton testID="viewIssuesButton" type="primary" text="Button" disabled onPress={() => {}} />);
  
    const textComponent = tree.root.findByProps({ testID: 'viewIssuesButton' }).props;
  
    expect(textComponent.disabled).toBeTruthy();
  });
  
  it('it renders button onPress', async () => {
    const onPressEvent = jest.fn();
    onPressEvent.mockReturnValue('Link on press invoked');
    
    const tree = create(<CustomButton testID="viewIssuesButton" type="primary" text="Button" onPress={onPressEvent} />);
    tree.root.findByProps({ testID: 'viewIssuesButton' }).props.onPress();
    
    expect(onPressEvent.mock.calls.length).toBe(1);
  });
  
  it('it renders button with leftIcon', async () => {
    const bookmarkIcon = <AntDesign size={20} color={Colors.blue} name={'star'} />;
    const tree = create(<CustomButton leftIcon={bookmarkIcon} testID="viewIssuesButton" type="primary" text="Button" onPress={() => {}} />).toJSON();
  
    expect(tree).toMatchSnapshot();
  });

  it('it renders button with rightIcon', async () => {
    const bookmarkIcon = <AntDesign size={20} color={Colors.blue} name={'star'} />;
    const tree = create(<CustomButton rightIcon={bookmarkIcon} testID="viewIssuesButton" type="primary" text="Button" onPress={() => {}} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
