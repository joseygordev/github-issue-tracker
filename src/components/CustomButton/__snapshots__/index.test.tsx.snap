// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton it renders button with leftIcon 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderColor": "#00aaff",
      "borderRadius": 15,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 42,
      "justifyContent": "center",
      "marginVertical": 8,
      "opacity": 1,
      "width": 120,
    }
  }
  testID="viewIssuesButton"
>
  <Text />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Array [
          Object {
            "color": "#00aaff",
            "fontSize": 12,
            "letterSpacing": 0.8,
          },
          Object {},
        ],
        Object {
          "fontFamily": "space-mono",
        },
      ]
    }
    testID="textValue"
  >
    BUTTON
  </Text>
</View>
`;

exports[`CustomButton it renders button with rightIcon 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderColor": "#00aaff",
      "borderRadius": 15,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 42,
      "justifyContent": "center",
      "marginVertical": 8,
      "opacity": 1,
      "width": 120,
    }
  }
  testID="viewIssuesButton"
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Array [
          Object {
            "color": "#00aaff",
            "fontSize": 12,
            "letterSpacing": 0.8,
          },
          Object {},
        ],
        Object {
          "fontFamily": "space-mono",
        },
      ]
    }
    testID="textValue"
  >
    BUTTON
  </Text>
  <Text />
</View>
`;
